go/controller/api/v1alpha1/deploymentsettings_types.go

```diff
@@ -176,6 +180,50 @@ func (r *ElasticsearchConnection) Attributes(ctx context.Context, c client.Clien
 	return attr, nil
 }
 
+type OpensearchConnection struct {
+	// Host ...
+	//
+	// +kubebuilder:validation:Required
+	Host string `json:"host"`
+
+	// Index to query in opensearch
+	//
+	// +kubebuilder:validation:Optional
+	Index string `json:"index"`
+
+	// AWS Access Key ID to use, can also use IRSA to acquire credentials
+	//
+	// +kubebuilder:validation:Optional
+	AWSAccessKeyID *string `json:"awsAccessKeyId,omitempty"`
+
+	// AWS Secret Access Key to use, can also use IRSA to acquire credentials
+	//
+	// +kubebuilder:validation:Optional
+	AwsSecretAccessKeyRef *corev1.SecretKeySelector `json:"awsSecretAccessKeyRef,omitempty"`
+
+	// AWS Region to use
+	//
+	// +kubebuilder:validation:Optional
+	AWSRegion *string `json:"awsRegion,omitempty"`
+}
+
+func (r *OpensearchConnection) Attributes(ctx context.Context, c client.Client, namespace string) (*console.OpensearchConnectionAttributes, error) {
+	attr := &console.OpensearchConnectionAttributes{
+		Host:           r.Host,
+		Index:          r.Index,
+		AWSAccessKeyID: r.AWSAccessKeyID,
+	}
+	if r.AwsSecretAccessKeyRef != nil {
+		secret := &corev1.Secret{}
+		if err := c.Get(ctx, types.NamespacedName{Name: r.AwsSecretAccessKeyRef.Name, Namespace: namespace}, secret); err != nil {
```

Comment:
There's an existing utility for this (utils.GetSecretKey, see the other invocations in this file).  We should also double check this still retries if not found.